tests/apollo/Apollo.test.ts

```diff
@@ -323,4 +323,157 @@ describe("Apollo", () => {
     expect(raw1).to.equal(raw2);
     expect(raw1).to.not.equal(raw3);
   });
+
+  describe("createPrivateKey", () => {
+    const seedHex = "947877896c61a5c64f266adbebbc69a2a01f1a2cfbf72c08a11c693d0429ccded34bdc0c28b5be910a5095b97e7bc6e3e209527ce8e75f9964d25cd6f6ad63e0";
+
+    describe("Secp256k1", () => {
+      it("default - creates a new key", () => {
+        const result = apollo.createPrivateKey({
+          [KeyProperties.type]: KeyTypes.EC,
+          [KeyProperties.curve]: Curve.SECP256K1,
+          [KeyProperties.seed]: seedHex,
+        });
+
+        expect(result).to.be.an.instanceOf(Secp256k1PrivateKey);
+        expect(result.raw).to.eql(Uint8Array.from([48, 82, 38, 112, 243, 185, 72, 236, 243, 178, 189, 199, 153, 90, 179, 73, 176, 178, 227, 149, 217, 24, 191, 123, 45, 65, 144, 242, 89, 29, 253, 162]));
+        expect(result.type).to.eq(KeyTypes.EC);
+        // ?? should this be set
+        // expect(result.getProperty(KeyProperties.type)).to.eq(KeyTypes.EC);
+
+        expect(result.curve).to.eq(Curve.SECP256K1);
+        expect(result.getProperty(KeyProperties.curve)).to.eq(Curve.SECP256K1);
+
+        // ? why don't we have `.seed` if we have `.type` and `.curve` properties
+        // expect(result.seed).to.eq(Curve.SECP256K1);
+        expect(result.getProperty(KeyProperties.seed)).to.eq(seedHex);
+
+        // no derivationPath provided, defaults to `m/0'/0'/0'`
+        // ??? why is derivationPath Hexed - should be DerivationPath instance or string
+        // expect(result.getProperty(KeyProperties.derivationPath)).to.eq(`m/0'/0'/0'`);
+        expect(result.getProperty(KeyProperties.derivationPath)).to.eq("6d2f30272f30272f3027");
```

Comment:
can someone explain why he hex encode the derivationPath?

Reply:
The origin i guess is because we at some point need to convert this into a format that can be later recovered from the database.

And because each field have a different type or may have it, basically. Buffers cannot be stored, etc